### PR TITLE
[Macros] Use SwiftCompilerPluginMessageHandling JSON encoder/decoder

### PR DESCRIPTION
**Pull request by rintaro**
* Replace `swiftLLVMSJON` with `SwiftCompilerPluginMessageHandling.JSON`
* Remove `swiftLLVMJSON`
* Use `SwiftCompilerPluginMessageHandling.StandardIOMessageConnection` in `swift-plugin-server`

### CODE REVIEW

**Comment by rintaro** [comment]
https://github.com/apple/swift-syntax/pull/2598
@swift-ci Please smoke test

**Comment by rintaro** [comment]
https://github.com/apple/swift-syntax/pull/2618
@swift-ci Please smoke test

**Comment by rintaro** [comment]
https://github.com/apple/swift-syntax/pull/2618
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Review by ahoppen — APPROVED** [review]

File: tools/swift-plugin-server/CMakeLists.txt [hunk 1]
```diff
@@ -1,8 +1,9 @@
 if (SWIFT_BUILD_SWIFT_SYNTAX)
   # _swiftCSwiftPluginServer is just a C support library for swift-plugin-server
   # Don't bother to create '.a' for that.
-  add_swift_host_library(_swiftCSwiftPluginServer OBJECT
+  add_swift_host_library(_swiftCSwiftPluginServer STATIC
```
Comment (hamishknight) [inline tools/swift-plugin-server/CMakeLists.txt:4]: Looks like the comment needs updating

**Review by hamishknight — APPROVED** [review]

**Pull request merged**